app/controlplane/plugins/core/guac/v1/guac.go

```diff
@@ -0,0 +1,268 @@
+}
+
+// Register is executed when a operator wants to register a specific instance of this integration with their Chainloop organization
+func (i *Integration) Register(ctx context.Context, req *sdk.RegistrationRequest) (*sdk.RegistrationResponse, error) {
+	i.Logger.Info("registration requested")
+
+	// Marshal the request information
+	var request *registrationRequest
+	if err := sdk.FromConfig(req.Payload, &request); err != nil {
+		return nil, fmt.Errorf("invalid registration request: %w", err)
+	}
+
+	// Check that the credentials are valid and the bucket exists
+	client, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(request.Credentials)))
+	if err != nil {
+		return nil, fmt.Errorf("creating storage client: %w", err)
+	}
+
+	_, err = loadBucket(ctx, client, request.Bucket)
+	if err != nil {
+		return nil, err
+	}
+
+	// Store the bucket and provider in the state
+	rawConfig, err := sdk.ToConfig(&registrationState{
+		Bucket:   request.Bucket,
+		Provider: string(providerGCS),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("marshalling configuration: %w", err)
+	}
+
+	return &sdk.RegistrationResponse{
+		// We also want to store the credentials securely
+		Credentials:   &sdk.Credentials{Password: request.Credentials},
+		Configuration: rawConfig,
+	}, nil
+}
+
+// Attachment is executed when to attach a registered instance of this integration to a specific workflow
+func (i *Integration) Attach(_ context.Context, _ *sdk.AttachmentRequest) (*sdk.AttachmentResponse, error) {
+	// Nothing to do during attachment
+	return &sdk.AttachmentResponse{}, nil
+}
+
+const filePrefix = "chainloop"
+
+// Execute will be instantiate when either an attestation or a material has been received
+// It's up to the plugin builder to differentiate between inputs
+func (i *Integration) Execute(ctx context.Context, req *sdk.ExecutionRequest) error {
+	i.Logger.Info("execution requested")
+
+	// Extract registration and attachment configuration if needed
+	var registrationConfig *registrationState
+	if err := sdk.FromConfig(req.RegistrationInfo.Configuration, &registrationConfig); err != nil {
+		return fmt.Errorf("invalid registration configuration %w", err)
+	}
+
+	client, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(req.RegistrationInfo.Credentials.Password)))
+	if err != nil {
+		return fmt.Errorf("creating storage client: %w", err)
+	}
+
+	bucket, err := loadBucket(ctx, client, registrationConfig.Bucket)
+	if err != nil {
+		return fmt.Errorf("loading bucket: %w", err)
+	}
+
+	// 1 - Upload the attestation
+	envelopeJSON, err := json.Marshal(req.Input.Attestation.Envelope)
+	if err != nil {
+		return fmt.Errorf("marshalling attestation: %w", err)
+	}
+
+	// chainloop-deadbeef-attestation.json
+	fileName := fmt.Sprintf("%s-%s-attestation.json", filePrefix, req.Input.Attestation.Hash.Hex)
+	if err := uploadToBucket(ctx, bucket, fileName, envelopeJSON, req.ChainloopMetadata, i.Logger); err != nil {
```

Comment:
Will Guac be able to process our attestations? Is it fully compatible?

Reply:
It is not. It will skip them for now. We should work with them and or the in-toto team to figure out if we can either contribute or adopt a standard predicate type. That effort is tracked here https://github.com/chainloop-dev/chainloop/issues/60